lib/material/material_gui.flow

```diff
@@ -1468,4 +1478,142 @@ MOverlayT(margin : Transform<double>, radius : Transform<double>, style : Transf
 		\sz -> MGetFocusGroup(\parent -> TFrameT(margin, radius, fselectLift(style, \s -> mgraphicsStyle2tgraphicsStyle(parent, s)), sz)),
 		true
 	)
+}
+
+MSegmentedButtons(
+	items : [SegmentButtonItem],
+	style : [SegmentButtonsStyle],
+	graphicsStyles : [MGraphicsStyle]
+) -> Material {
+	MGetFocusGroup(\fg -> {
+		multiSelect = extractStruct(style, MMultiSelect(false)).enabled;
+		iconsOnly = extractStruct(style, MIconsOnly(false)).enabled;
+		defaultBgColor = extractStruct(graphicsStyles, MThemedFill(MSecondaryColor())).color;
+		defaultStrokeColor = extractStruct(graphicsStyles, MThemedStroke(either(fg.theme.focusOutlineColor, MAccentColor()))).color;
+		
+		paddingWithoutIcon = 12.0;
+
+		maxHeight = make(0.0);
+		lastSelected = make(None());
+
+		makeButton = \i, item -> {
+			buttonState = item.state;
+			isSelected = item.selected;
+			maxWidth = make(60.0);
```

Comment:
Don't understand what are you trying to do with maxWidth. AFAICS, you creating a loop: maxWidth affects MCenterIn block width and then you are inspecting it. That's why you cant shrink elements after expanding.

Also, 60 as hardcoded initial value looks suspicious.

Try to rework using fillers.